### PR TITLE
On OSX incorporate nano time in wakeUpASyncReporter sem name

### PR DESCRIPTION
**Pull request by pshipton**
Issue https://github.com/eclipse-openj9/openj9/issues/12526

### CODE REVIEW

**Comment by pshipton** [comment]
@keithc-ca can you pls take a look.

File: port/unix/omrsignal.c
Comment (keithc-ca) [inline port/unix/omrsignal.c:1593]: Instead of `128`, it should use `sizeof(semName)`.

If PIDs are 32-bits, this will be at most 30 characters, but if PIDs can be larger than 2^32, then it could be up to 39 characters in which case the prefix needs to shrink or some other strategy needs to be used to guarantee uniqueness.

If we use hex instead, the maximum lengths are smaller, but still potentially problematic if a PID needs a full 64 bits.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: port/unix/omrsignal.c
Comment (pshipton) [reply port/unix/omrsignal.c]: sizeof(pid_t) is 4

Comment (pshipton) [reply port/unix/omrsignal.c]: Any idea why using just the pid sometimes gives a "File exists" perror?

File: port/unix/omrsignal.c
Comment (pshipton) [reply port/unix/omrsignal.c]: Only one problem
```
13:39:45  /Users/jenkins/workspace/Build_JDK8_x86-64_mac_Personal/omr/port/unix/omrsignal.c:1557:15: error: use of undeclared identifier 'SEM_NAME_LEN'
13:39:45          char semName[SEM_NAME_LEN + 1];
```
I searched the include files and didn't find it defined. I can hard code 32.

Comment (keithc-ca) [reply port/unix/omrsignal.c]: Perhaps `#include <semaphore.h>` would help.

Comment (pshipton) [reply port/unix/omrsignal.c]: It's there, see line 46

Comment (keithc-ca) [reply port/unix/omrsignal.c]: Line 46 is only for z/OS.

Comment (keithc-ca) [reply port/unix/omrsignal.c]: I'm ok with hard-coding it:
```
char semName[31 /* SEM_NAME_LEN */ + 1];
```

File: port/unix/omrsignal.c
Comment (keithc-ca) [reply port/unix/omrsignal.c]: Sorry, no theories about that.

File: port/unix/omrsignal.c
Comment (keithc-ca) [inline port/unix/omrsignal.c:1593]: I don't know if this implies the creation of a directory (/omr) internally: maybe there's a race doing that. These semaphores aren't visible in the file-system so `"/omr-wUASRptr%x-%x"` might be a better pattern in that it doesn't present the opportunity for such a race condition.

Comment (pshipton) [reply port/unix/omrsignal.c]: Works for me, I've made that change.

Comment (pshipton) [reply port/unix/omrsignal.c]: Started some testing via https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/69/

Comment (pshipton) [reply port/unix/omrsignal.c]: Pushing another change caused the build to fail, trying again.
https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/70/

Comment (pshipton) [reply port/unix/omrsignal.c]: The OpenJ9 sanity.functional testing passed.

File: port/unix/omrsignal.c
Comment (keithc-ca) [inline port/unix/omrsignal.c:1593]: The tiniest of suggestions: remove 'ptr' from the name, the rest of 'wuasr' is an acronym.

Comment (pshipton) [reply port/unix/omrsignal.c]: Like so?

Comment (keithc-ca) [reply port/unix/omrsignal.c]: Yes.

Comment (pshipton) [reply port/unix/omrsignal.c]: I don't think there is anything left to address, so waiting for your approval or further review comments.

**Comment by pshipton** [comment]
jenkins build all

**Review by keithc-ca — APPROVED** [review]

**Comment by pshipton** [comment]
@0xdaryl can you pls take a look.

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**
